### PR TITLE
feat/ls

### PR DESCRIPTION
**Pull request by diasdavid**
https://github.com/diasdavid/registry-mirror/issues/8

### CODE REVIEW

File: commands/ls.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      })
+    } else {
+      ls()
+    }
+
+    function ls () {
+      var apiCtl = ipfsAPI('/ip4/127.0.0.1/tcp/5001')
+      apiCtl.files.ls('/npm-registry', function (err, res) {
+        if (err) {
+          return console.log(err)
```
Comment (Dignifiedquire) [inline commands/ls.js]: Why not `console.error`?

File: commands/ls.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  run: function (update, name) {
+    if (update) {
+      fetchIPNS({ blobStore: true }, function () {
+        ls()
+      })
+    } else {
+      ls()
+    }
+
+    function ls () {
```
Comment (Dignifiedquire) [inline commands/ls.js]: Why define this inside this object, you could just pull this out to the top

File: src/fetch-ipns.js [hunk 3]
```diff
@@ -38,6 +38,9 @@ function fetchIPNS (config) {
                 return logger.err('Failed to clone /npm-registry mDAG node', err)
               }
               logger.info('Updated directory listing, good to go :)')
+              if (callback) {
```
Comment (Dignifiedquire) [inline src/fetch-ipns.js]: Shouldn't this pass errors through?

**Comment by diasdavid** [comment]
thank you for CR @Dignifiedquire :D

**Comment by Dignifiedquire** [comment]
@diasdavid always happy to tell you what you did wrong :stuck_out_tongue_closed_eyes:

**Pull request merged**